MediaBrowser.Providers/MediaInfo/FFProbeVideoInfo.cs

```diff
@@ -526,16 +529,21 @@ private SubtitleOptions GetOptions()
         /// <param name="options">The refreshOptions.</param>
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>Task.</returns>
-        private async Task AddExternalSubtitles(
+        private async Task AddExternalSubtitlesAsync(
             Video video,
             List<MediaStream> currentStreams,
             MetadataRefreshOptions options,
             CancellationToken cancellationToken)
         {
-            var subtitleResolver = new SubtitleResolver(_localization);
-
             var startIndex = currentStreams.Count == 0 ? 0 : (currentStreams.Select(i => i.Index).Max() + 1);
-            var externalSubtitleStreams = subtitleResolver.GetExternalSubtitleStreams(video, startIndex, options.DirectoryService, false);
+            var externalSubtitleStreamsAsync = _subtitleResolver.GetExternalStreams(video, startIndex, options.DirectoryService, false, cancellationToken);
+
+            List<MediaStream> externalSubtitleStreams = new List<MediaStream>();
+
+            await foreach (MediaStream externalSubtitleStream in externalSubtitleStreamsAsync)
+            {
+                externalSubtitleStreams.Add(externalSubtitleStream);
+            }
```

Comment:
Use `ToListAsync()` instead?

Reply:
`'IAsyncEnumerable<MediaStream>' does not contain a definition for 'ToListAsync'`

Reply:
https://www.nuget.org/packages/System.Linq.Async

Reply:
I'm not sure there's really any reason to use IAsyncEnumerable though. It's still too new to be really useful for small stuff like this